### PR TITLE
Add rule for restricted patterns of Keywords Given, When, Then

### PR DESCRIPTION
**Pull request by AlexisArteaga13**
The validation of the patterns of the keywords given, when and then is added, in addition to the update of its unit tests and the modification of the readme.

### CODE REVIEW

File: src/rules/no-restricted-patterns.ts [hunk 1]
```diff
@@ -1,36 +1,43 @@
...
   'Background': [] as string[],
   'Feature': [] as string[],
+  'Given': [] as string[],
+  'When': [] as string[],
+  'Then': [] as string[]
 } as Configuration;
 
-export function run({feature}: GherkinData, configuration: Configuration): RuleError[] {
+export function run({ feature }: GherkinData, configuration: Configuration): RuleError[] {
+  previosKeyword = '';
```
Comment (Nyaran) [inline src/rules/no-restricted-patterns.ts:34]: wrong spelling
```suggestion
  previousKeyword = '';
```

File: src/rules/no-restricted-patterns.ts [hunk 2]
```diff
@@ -1,36 +1,43 @@
...
   Scenario: T[]
   ScenarioOutline: T[]
   Background: T[]
   Feature: T[]
+  Given: T[]
+  When: T[]
+  Then: T[]
 };
 
+const Keywords = ['Given', 'When', 'Then'];
```
Comment (Nyaran) [inline src/rules/no-restricted-patterns.ts:16]: Please keep coding style consistency, variables must be in camelCase, starting with lowercase
```suggestion
const keywords = ['Given', 'When', 'Then'];
```

**Review by Nyaran — CHANGES_REQUESTED** [review]
Looks good, just some small fixes needed.

Also I think this change could be improved if you add the type "Step", that would apply to all kind of steps, allowing to simplify the rule config if someone wants to ban a pattern to all kind of steps.

**Comment by AlexisArteaga13** [comment]
> Looks good, just some small fixes needed.
> 
> Also I think this change could be improved if you add the type "Step", that would apply to all kind of steps, allowing to simplify the rule config if someone wants to ban a pattern to all kind of steps.
Excellent, I check it.

**Comment by AlexisArteaga13** [comment]
Buenas, tengo resueltas las observaciones pero está apareciendo este error: 
image

Alguna idea de como solucionarlo? me apareció después de hacer un git pull.

File: src/rules/no-restricted-patterns.ts [hunk 3]
```diff
@@ -40,34 +49,44 @@ export function run({feature}: GherkinData, configuration: Configuration): RuleE
...
   return restrictedPatterns;
 }
 
 function getRestrictedPatternsForNode(node: GherkinNode, restrictedPatterns: ConfigurationPatterns, language: string): RegExp[] {
   const key = gherkinUtils.getLanguageInsensitiveKeyword(node, language).toLowerCase() as keyof ConfigurationPatterns;
-
+  if (keywords.map(item => item.toLowerCase()).includes(key.toLowerCase())) {
+    previousKeyword = key;
+  }
+  if (key.toLowerCase() === 'and' && keywords.map(item => item.toLowerCase()).includes(previousKeyword.toLowerCase())) {
```
Comment (Nyaran) [inline src/rules/no-restricted-patterns.ts:87]: This only affects to 'And' keywords, but has no effect for 'or' or "*"

**Review by Nyaran — CHANGES_REQUESTED** [review]

File: src/rules/no-restricted-patterns.ts (hunk 3 above)
Comment (AlexisArteaga13) [reply src/rules/no-restricted-patterns.ts]: Hello, I had the problem that I mentioned above, about the library, is it solved? In order to review it, I can't run the tests .

**Comment by Nyaran** [comment]
> Buenas, tengo resueltas las observaciones pero está apareciendo este error: image
> 
> Alguna idea de como solucionarlo? me apareció después de hacer un git pull.

Solucionado con [2ba72c7](https://github.com/Nyaran/gplint/pull/433/commits/2ba72c7b347d44f6a9965bec08e3f160ad359601)

**Review by Nyaran — APPROVED** [review]

**Comment by Nyaran** [comment]
@AlexisArteaga13 Avísame cuando confirmes que te funciona para integrar los cambios y generar nueva versión

**Comment by AlexisArteaga13** [comment]
> @AlexisArteaga13 Avísame cuando confirmes que te funciona para integrar los cambios y generar nueva versión

Confirmado !! @Nyaran

**Pull request merged**